### PR TITLE
feat: Add mentions instead of search

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We are going to replace the search with cmd+k search using the nice unified search. Until then, I think we should ditch backlinks and instead have mentions. 

## Changes

* Removes backlinks (we can add it back in when we have a solid search)
* Adds "mentions"
* Moves membersLogic out of settings as it used everywhere
* Fixes to not show the share button for local notebooks

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeMention.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+export const NotebookNodeMention = Node.create({
+    name: NotebookNodeType.Mention,
+    inline: true,
+    group: 'inline',
+    atom: true,
+
+    serializedText: (attrs: NotebookNodeMentionAttrs): string => {
+        // timestamp is not a block so `getText` does not add a separator.
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeMention.tsx:48]:
```suggestion
        // mention is not a block so `getText` does not add a separator.
```

File: frontend/src/scenes/notebooks/Notebook/Editor.tsx [hunk 2]
```diff
@@ -29,7 +29,6 @@ import { NotebookNodeImage } from '../Nodes/NotebookNodeImage'
 
 import { EditorFocusPosition, EditorRange, JSONContent, Node, textContent } from './utils'
 import { SlashCommandsExtension } from './SlashCommands'
-import { BacklinkCommandsExtension } from './BacklinkCommands'
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Notebook/Editor.tsx:32]: I'd be in favour of deleting this file if we're not using it. Or leaving it and just changing the key mapping to `char: '#',` or similarly appropriate character

File: frontend/src/scenes/notebooks/Notebook/MentionsExtension.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    )
+})
+
+const MentionsPopover = forwardRef<MentionsRef, MentionsPopoverProps>(function MentionsPopover(
+    { visible = true, decorationNode, children, onClose, ...props }: MentionsPopoverProps,
+    ref
+): JSX.Element | null {
+    return (
+        <Popover
+            placement="right-start"
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Notebook/MentionsExtension.tsx:146]: Tiny thing but it felt strange for the popover to appear right next to the `@` symbol. Double checked GitHub and Slack which both go above or below but really up to you

**Review by daibhin — APPROVED** [review]

File: frontend/src/scenes/notebooks/Notebook/Editor.tsx (hunk 2 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/Editor.tsx:32]: Tempted to keep it removed as it is super buggy tbh. It steals mouse focus and sometimes gets stuck in a weird state.

Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/Editor.tsx:32]: Then we can come back to it later when we have the new search

**Pull request merged**